lib/BackgroundJob/ExpireGroupVersions.php

```diff
@@ -42,7 +50,42 @@ public function __construct(GroupVersionsExpireManager $expireManager, ITimeFact
 		$this->expireManager = $expireManager;
 	}
 
+	/**
+	 * Expiring groupfolder versions can be quite expensive.
+	 * We need to limit the amount of folders we expire per run.
+	 */
 	protected function run($argument) {
-		$this->expireManager->expireAll();
+		$lastFolder = $this->appConfig->getValueInt(Application::APP_ID, 'cron_last_folder_index', 0);
+		$folders = $this->folderManager->getAllFolders();
+
+		$folderCount = count($folders);
+		$currentRunHour = (int)date('G', $this->timeFactory->getTime());
+
+		// Calculate folders to process in the remaining hours, ensure at least one folder is processed
+		$toDo = max(1, (int)ceil(($folderCount - $lastFolder) / (24 - $currentRunHour)));
+
+		// If there are no folders, we don't need to do anything
+		if ($folderCount === 0) {
+			$this->logger->debug('No folders to expire', ['app' => 'cron']);
+			return;
+		}
+
+		// If we would go over the end of the list, wrap around
+		if ($lastFolder + $toDo >= $folderCount) {
+			$lastFolder = 0;
+		}
```

Comment:
If `true`, then anything between `$lastFolder` and `$folderCount` will not be expired, no?

Reply:
Hmm, good point, let me adjust the logic

Reply:
Done